flocker/control/schema/types.yml

```diff
@@ -59,7 +59,17 @@ definitions:
         external:
           type: integer
     uniqueItems: true
-    
+
+  environment:
+    title: "Container environment variables"
+    description: "Mapping of label and value pairs for environment variables"
+    type: object
+    patternProperties:
+      "^.{1,256}$":
```

Comment:
Yeah, ok. There doesn't seem to be any standard so I suppose it's fine to let the user supply that they like.

http://stackoverflow.com/questions/2821043/allowed-characters-in-linux-environment-variable-names

But I suppose it's also limited to characters that are allowed in JSON keys...I'm not sure what that is.

The 256 character limit may be a little low though, especially for environment variable values which may contain things like public ssh keys.